erdjs/src/message/message.ts

```diff
@@ -0,0 +1,79 @@
+import { sign } from "tweetnacl";
+import { Address } from "../address";
+import { ISignable } from "../interface";
+import { Signature } from "../signature";
+import { MessageVersion, MessageSigner } from "./messageParams";
+
+const MESSAGE_VERSION = new MessageVersion(1);
+const MESSAGE_PREFIX = "\x17Elrond Signed Message:\n";
+
+export class Message implements ISignable {
+
+    /**
+     * Signer of the message.
+     */
+    address: Address;
+
+    /**
+     * Actual message being signed.
+     */
+    value: Buffer;
+
+    /**
+     * Signature obtained by a signer of type @param signer .
+     */
+    signature: Signature;
+
+    /**
+     * Message version code in case we modify this schema.
+     */
+    version: MessageVersion;
+
+    /**
+     * Specifies which ISigner is to be used to sign and verify this message.
+     */
+    signer: MessageSigner;
+
+    public constructor(init?: Partial<Message>) {
+        this.address = Address.Zero();
+        this.value = Buffer.from([]);
+        this.signature = new Signature();
+        this.version = MESSAGE_VERSION;
+        this.signer = MessageSigner.Wallet;
+
+        Object.assign(this, init);
+    }
+
+    serializeForSigning(_signedBy: Address): Buffer {
+        return Buffer.concat([Buffer.from(MESSAGE_PREFIX), this.value]);
```

Comment:
If we only use the value for message signing, why do we need the rest of the metadata?

Reply:
They're useful so you can serialize the entire `Message` object and send it wherever you need. It encapsulates the signing / verifying required data:
- `address` is required for signature verification
- `value` is, as you said, the effectively signed data
- `signature` is the signed `value` (prefix added)
- `version` is there for future-proofing, in case we change stuff (prefix, message structure etc)
- `signer` is used to see what kind of protocol was used for signing the message (and subsequently what to use to verify the signature). I made no assumption that every signature will use the same ed25519 algo in the future.

Let me know what you think!